tests/unit/amo/utils/test_index.js

```diff
@@ -182,4 +183,70 @@ describe(__filename, () => {
       );
     });
   });
+
+  describe('stripLangFromAmoUrl', () => {
+    // These are known values from the config.
+    const validLang = 'en-US';
+    const invalidLang = 'en_US';
+
+    it('should return the same url if the link is external', () => {
+      const _checkInternalURL = sinon
+        .stub()
+        .returns({ isInternal: false, relativeURL: '' });
+      const urlString = `https://somehost/${validLang}/somepath/`;
+      expect(stripLangFromAmoUrl({ _checkInternalURL, urlString })).toEqual(
+        urlString,
+      );
+    });
+
+    it('should strip the lang the link is internal', () => {
+      const _checkInternalURL = sinon
+        .stub()
+        .returns({ isInternal: true, relativeURL: '' });
+      const urlString = `https://somehost/${validLang}/somepath/`;
+      expect(stripLangFromAmoUrl({ _checkInternalURL, urlString })).toEqual(
+        'https://somehost/somepath/',
+      );
+    });
+
+    it('should not strip an invalid lang', () => {
```

Comment:
Are you sure?

Reply:
Well, I was basing this on our ability to recognize a lang. If we're not only going to remove valid langs, how will we know if we should remove something or not?